server/container/guice/guice-common/src/test/java/org/apache/james/modules/ConfigurationsPerformerTest.java

```diff
@@ -0,0 +1,171 @@
+            Multibinder.newSetBinder(binder(), ConfigurationPerformer.class).addBinding().to(AConfigurationPerformer.class);
+        }
+    }
+
+    private static class AConfigurationPerformer implements ConfigurationPerformer {
+
+        private final A a;
+
+        @Inject
+        private AConfigurationPerformer(A a) {
+            this.a = a;
+        }
+
+        @Override
+        public void initModule() {
+            try {
+                a.configure(null);
+            } catch (ConfigurationException e) {
+                Throwables.propagate(e);
+            }
+        }
+
+        @Override
+        public List<Class<? extends Configurable>> forClasses() {
+            return ImmutableList.of(A.class);
+        }
+    }
+
+    private static class BConfigurationPerformer implements ConfigurationPerformer {
+
+        private final B b;
+
+        @Inject
+        private BConfigurationPerformer(B b) {
+            this.b = b;
+        }
+
+        @Override
+        public void initModule() {
+            try {
+                b.configure(null);
+            } catch (ConfigurationException e) {
+                Throwables.propagate(e);
+            }
+        }
+
+        @Override
+        public List<Class<? extends Configurable>> forClasses() {
+            return ImmutableList.of(B.class);
+        }
+    }
+
+    @Test
+    public void initModulesShouldBePerformedOneTimeWhenConfigurableModuleContainsMultipleDependencies() throws Exception {
+        Injector injector = Guice.createInjector(new ConfigurablesModule(),
+                new ABConfigurationPerformerModule());
+
+        injector.getInstance(ConfigurationsPerformer.class).initModules();
+
+        assertThat(injector.getInstance(A.class).isConfigured()).isTrue();
+        assertThat(injector.getInstance(B.class).isConfigured()).isTrue();
+    }
+
+    private static class ABConfigurationPerformer implements ConfigurationPerformer {
+
+        private final A a;
+        private final B b;
+        private boolean configured;
+
+        @Inject
+        private ABConfigurationPerformer(A a, B b) {
+            this.a = a;
+            this.b = b;
+            this.configured = false;
+        }
+
+        @Override
+        public void initModule() {
+            if (configured) {
+                throw new RuntimeException("Already configured");
+            }
+
+            try {
+                a.configure(null);
+                b.configure(null);
+                configured = true;
+            } catch (ConfigurationException e) {
+                Throwables.propagate(e);
+            }
+        }
+
+        @Override
+        public List<Class<? extends Configurable>> forClasses() {
+            return ImmutableList.of(A.class, B.class);
+        }
+    }
+
+    private static class ABConfigurationPerformerModule extends ConfigurablesModule {
```

Comment:
what is the difference with "MyModule" ? What about giving meaningful names to these classes ?